src/components/NetworkAssetSelectInput/NetworkAssetSelectInput.tsx

```diff
@@ -320,38 +325,92 @@ const NetworkAssetSelectInput = ({
       await getSmartWalletBalancesByChain(walletAddress, supportedChains);
     };
     handleBalanceGet();
-  }, [supportedChains, sdk, walletAddress]);
+  }, [supportedChains, sdk, accountAddress]);
+
+  useEffect(() => {
+    const handleKeybasedBalanceGet = async () => {
+      if (!sdk || !providerAddress) return;
+      await getKeybasedWalletBalancesPerChain(providerAddress, supportedChains);
```

Comment:
I'm thinking if these shouldn't go somwhere in the upper level like `EtherspotContextProvider` because everytime we will need balances we have to add these lines, right? wdyt

Reply:
Do you mean  `EtherspotContextProvider` or `TransactionBuilderContextProvider`?

Reply:
`EtherspotContextProvider` because it holds values for accounts generally